### ISSUE TITLE
Predict data trends

### ISSUE THREAD

**Issue by SlavaDobr** [yhwang/node-red-contrib-tf-model#10]
Dear @yhwang !
Thanks for sharing this node!

I tried to get a simple **linear regression** in my flow.
But when I loaded my test model in JSON format, the node threws an error: "Cannot read property 'producer' of undefined".
I searched the internet. As I understand it, the problem is loading the model, but here the node controls it.
I am not a JavaScript developer.
In fact, I need to get readings from sensors, and then predict trends.
Can I do it using this node?
Can you help me with an example please?

Thanks a lot in advance.

Best Regards!
Slava.

**Comment by yhwang** [comment]
can you share you model and I may try it?

**Comment by SlavaDobr** [comment]
Thanks for the reply!
Yes of course. I apologize in advance for the such simple model. It seems to me that it should have a different look, but I do not know which one.
I am new to TensorFlow, so I made an example from this video: https://youtu.be/pbCExciEbrc?t=336.
This example worked in a browser.
So I thought to check it in the node.
Here is my code to get the model:

`async function learnLinear(){
   const model = tf.sequential();
   model.add(tf.layers.dense({units : 1, inputShape: [1]}));
    model.compile({
       loss: 'meanSquaredError',
       optimizer: 'sgd'
   });
   const xs = tf.tensor2d([-1, 0, 1, 2, 3, 4], [6, 1]);
   const ys = tf.tensor2d([-3, -1, 1, 3, 5, 7], [6, 1]);
   await model.fit(xs, ys, {epochs : 250});
   await model.save('downloads://IIoT');
}

learnLinear();`

As a result, I get a model file in JSON format and a weights file for this model.
[IIoT.zip](https://github.com/yhwang/node-red-contrib-tf-model/files/4613326/IIoT.zip)

After deploying the flow with this model, the indicated error is thrown.

**Comment by yhwang** [comment]
ah, I see the problem. Originally, I created this node-red custom node to load `Graph model` and the model you stored is in `Layer model` format. I think your usage is valid and I'd like to support `Layer model` as well. I will do some quick testing tomorrow and see if I can support `Layer model` quickly.

**Comment by SlavaDobr** [comment]
Thank you so much!
I will look forward to the update :)

**Comment by yhwang** [comment]
the tf-model node supports LayersModel now. However, the payload passing to the tf-model using LayersModel would be `tf.Tensor | tf.Tensor[]`. I would update the documentation later.

I put a `tf function` node with the following code:
```
return {payload: tf.tensor1d([1])};
```
and linked it to a tf-model node with the model you provided. And it gave me the result:
```
{payload: [[1.0649425983428955]]}
```

**Comment by SlavaDobr** [comment]
Thank you so much!
It worked!)
Good luck and good remote work :)
